.github/workflows/tests.yml

```diff
@@ -0,0 +1,143 @@
+
+      - name: Check for invalid UUIDs
+        # can be removed once `configlet lint` gains this ability.
+        # Check issue https://github.com/exercism/configlet/issues/99
+        run: |
+          uuids=$(jq --raw-output '.exercises | map(.uuid) | .[]' config.json)
+          bad_uuid=$(echo "$uuids" | grep -vE '^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$' || true)
+          if [ -n "$bad_uuid" ]; then
+            echo "invalid UUIDs found! please correct these to be valid UUIDs:"
+            echo "$bad_uuid"
+            exit 1
+          fi
+
+  exercises:
+    name: Check exercises
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        resolver:
+          - lts-from-exercises
+          - nightly
+    # Discussion seems to indicate that continue-on-error is not exactly equivalent to Travis's allow_failures, but it's the best we can do for now.
+    # If failures on nightly cause too much confusion in CI status,
+    # consider just not running against the nightly resolver.
+    # https://github.com/actions/toolkit/issues/399
+    continue-on-error: ${{ matrix.resolver == 'nightly' }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install yq (for stack resolvers)
+        run: |
+         sudo add-apt-repository -y ppa:rmescandon/yq
+         sudo apt-get -q update
+         sudo apt-get -y install yq
+
+      - name: Resolve resolver
+        id: resolve-resolver
+        run: |
+          # We need this in two places:
+          # 1. cache key
+          # 2. --resolver arg to installing hlint
+          # It's unsafe to just use lts-from-exercises as the cache key,
+          # because that would cause caches from differing LTSes to collide
+          # (if we have updated the LTS version of all exercises between job runs).
+          if [ "${{ matrix.resolver }}" = "lts-from-exercises" ]; then
+            resolver=$(yq read "$(ls -1 exercises/*/stack.yaml | head -1)" resolver)
+          else
+            resolver=${{ matrix.resolver }}
+          fi
+          echo "::set-output name=resolver::$resolver"
+
+      - name: Week number
+        id: week-no
+        run: |
+          # We use this to form part of the cache key.
+          # This will get us to periodically rebuild from scratch,
+          # just to make sure that everything still works from scratch.
+          # Because we also have a weekly build at the start of every week,
+          # most of the time caches should hit.
+          echo "::set-output name=week-no::$(date -u +%Y-%U)"
+
+      - uses: actions/cache@v2
+        id: cache
+        with:
+          path: |
+            ~/.stack
+            ~/.foldercache
+            ~/.local/bin/hlint
+          key: ${{ steps.resolve-resolver.outputs.resolver }}-${{ steps.week-no.outputs.week-no }}-${{ hashFiles('exercises/**/package.yaml') }}
+          restore-keys: ${{ steps.resolve-resolver.outputs.resolver }}-${{ steps.week-no.outputs.week-no }}-
```

Comment:
> with restore-keys, changing one exercise's package.yaml will start from the previous cache and save a new cache

I understand this now, too. Thanks!

IIRC, when a restore-key only uses `resolver` and `week-no`, any changes that accumulate over one week without resolver updates, beyond the first change, won't get cached until the week later. This is probably acceptable.

Even though https://docs.github.com/en/free-pro-team@latest/actions/guides/caching-dependencies-to-speed-up-workflows has one example of

```
restore-keys: |
  npm-foobar-${{ hashFiles('package-lock.json') }}
  npm-foobar-
  npm-
```

I can't imagine how the first restore-key would work if `hashFiles` hits the new file(s) and not the old file(s). One could create an identifier based on the previous hash using `git` commands, rather than the current hash to create a chain of hashes.

Reply:
I forgot to say: ...but I don't think we need to do that at this point.

Reply:
>  any changes that accumulate over one week without resolver updates, beyond the first change, won't get cached until the week later. This is probably acceptable.

That is also my understanding, and I too think we can accept that.